osbs/build/user_params.py

```diff
@@ -312,60 +267,54 @@ class BuildUserParams(BuildCommon):
-        self.platform_node_selector = None
-        self.scratch_build_node_selector = None
-
-        self.required_params.extend([
-            self.build_type,
-            self.git_ref,
-            self.git_uri,
-        ])
-        self.repo_info = None
-
-        self.attrs_finalizer()
+    additional_tags = BuildParam("additional_tags")
+    base_image = BuildParam("base_image")
+    build_type = BuildParam("build_type", required=True)
+    compose_ids = BuildParam("compose_ids")
+    customize_conf = BuildParam("customize_conf", default=DEFAULT_CUSTOMIZE_CONF)
+    dependency_replacements = BuildParam("dependency_replacements")
+    filesystem_koji_task_id = BuildParam("filesystem_koji_task_id")
+    flatpak = BuildParam("flatpak", default=False)
+    git_branch = BuildParam("git_branch")
+    git_commit_depth = BuildParam("git_commit_depth")
+    git_ref = BuildParam("git_ref", default=DEFAULT_GIT_REF, required=True)
+    git_uri = BuildParam("git_uri", required=True)
+    imagestream_name = BuildParam("imagestream_name")
+    include_koji_repo = BuildParam("include_koji_repo", default=False)
+    is_auto = BuildParam("is_auto")
+    isolated = BuildParam("isolated")
+    kind = BuildParam("kind", default=KIND)
+    koji_parent_build = BuildParam("koji_parent_build")
+    koji_upload_dir = BuildParam("koji_upload_dir")
+    name = BuildIDParam()
+    operator_bundle_replacement_pullspecs = BuildParam("operator_bundle_replacement_pullspecs")
+    operator_manifests_extract_platform = BuildParam("operator_manifests_extract_platform")
+    parent_images_digests = BuildParam("parent_images_digests")
+    platforms = BuildParam("platforms")
+    release = BuildParam("release")
+    remote_source_build_args = BuildParam("remote_source_build_args")
+    remote_source_configs = BuildParam("remote_source_configs")
+    remote_source_url = BuildParam("remote_source_url")
+    skip_build = BuildParam("skip_build")
+    tags_from_yaml = BuildParam("tags_from_yaml")
+    trigger_imagestreamtag = BuildParam("trigger_imagestreamtag")
+    triggered_after_koji_task = BuildParam("triggered_after_koji_task")
+    yum_repourls = BuildParam("yum_repourls")
+
+    # TODO: see if these can be allowed to appear in JSON
+    auto_build_node_selector = BuildParam("auto_build_node_selector",
+                                          include_in_json=False)
+    explicit_build_node_selector = BuildParam("explicit_build_node_selector",
+                                              include_in_json=False)
+    isolated_build_node_selector = BuildParam("isolated_build_node_selector",
+                                              include_in_json=False)
+    platform_node_selector = BuildParam("platform_node_selector",
+                                        include_in_json=False)
+    scratch_build_node_selector = BuildParam("scratch_build_node_selector",
+                                             include_in_json=False)
```

Comment:
Previously, these were not defined as `BuildParam` => they would not be included in `to_json()` output. Unlike `repo_info`, node selectors have JSON-compatible values. Is there a reason why they should not appear in JSON?

Reply:
Don't include it into user_params. We should have it in BuildRequest directly

Reply:
Keep it as is here for now. (No export to json)